plugin/storage/integration/badgerstore_test.go

```diff
@@ -55,20 +56,14 @@ func (s *BadgerIntegrationStorage) initialize(t *testing.T) {
 	s.SkipArchiveTest = true
 }
 
-func (s *BadgerIntegrationStorage) clear() error {
-	return s.factory.Close()
-}
-
 func (s *BadgerIntegrationStorage) cleanUp(t *testing.T) {
-	err := s.clear()
-	require.NoError(t, err)
-	s.initialize(t)
+	s.factory.Purge()
 }
 
 func TestBadgerStorage(t *testing.T) {
 	SkipUnlessEnv(t, "badger")
 	s := &BadgerIntegrationStorage{}
 	s.initialize(t)
 	s.RunAll(t)
-	defer s.clear()
+	defer s.factory.Close()
```

Comment:
this is the wrong way to use defer - if RunAll panics you will never reach this point. It can be called right after `initialize()`, but even better is to add it as a `t.CleanUp()` callback inside `initialize()`.